sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -456,11 +458,83 @@ public Query apply(Function input, Context context) throws IOException {
 
         class NotQuery implements FunctionToQuery {
 
+            private class SymbolToNotNullContext {
+                private final HashSet<Reference> references = new HashSet<>();
+                boolean containsNullCanMatchFunction = false;
+
+                boolean add(Reference symbol) {
+                    return references.add(symbol);
+                }
+
+                Set<Reference> references() {
+                    return references;
+                }
+            }
+
+            private class SymbolToNotNullRangeQueryArgs extends SymbolVisitor<SymbolToNotNullContext, Void> {
+
+                private final Set<String> NULL_CAN_MATCH_FUNCTIONS = ImmutableSet.of("any", "coalesce");
+
+                private boolean nullCanMatch(Function symbol) {
+                    return NULL_CAN_MATCH_FUNCTIONS.contains(symbol.info().ident().name().toLowerCase());
```

Comment:
i think the ident is already lowercased in the analyzer, so no need to do it here again